src/UserFunctions.java

```diff
@@ -121,22 +146,37 @@ private void printCommands() {
         System.out.println(help); 
     }
 
-    private void createNewPuzzle(String seedWord) {
+    private void createNewPuzzle(String seedWord, char requiredLetter) {
+        System.out.println("Generating new puzzle ...");
+
+        String newLine = getNewLineCharacter();
+        try {
+            FileReader dictionaryFile = new FileReader(DICTIONARY_PATH);
+            puzzle = Puzzle.fromWord(seedWord, requiredLetter, dictionaryFile);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage() + newLine + 
+                               "Puzzle not generated. Please try again.");
+        } catch (FileNotFoundException e) {
+            System.out.println("A dictionary file could not be found." + 
+                               "Puzzle not generated.");
+        } catch (IOException e) {
+            System.out.println(e.getMessage() + newLine + "Puzzle not generated.");
+        }
+    }
+
+    private void createNewPuzzle() {
         System.out.println("Generating new puzzle ...");
 
         String newLine = getNewLineCharacter();
         try {
             FileReader dictionaryFile = new FileReader(DICTIONARY_PATH);
```

Comment:
We could store these FileReaders in global variables so that we don't have to make new ones every time the createNewPuzzle function is called. I don't know what the overhead of making a FileReader is, so I don't know if it would make it that much faster, but it would make fewer objects for the GC to clean up.

Reply:
I don't think it would save that much preformance, but it would move all file not found errors up to, likely, object construction rather than these function. May be worth implementing at some point, but not urgent, as it would require some changes to the class and constructor.

Reply:
Yeah, I don't think it's a big deal.